xmr-btc/src/alice/message.rs

```diff
@@ -0,0 +1,112 @@
+use anyhow::Result;
+use ecdsa_fun::{adaptor::EncryptedSignature, Signature};
+use std::convert::TryFrom;
+
+use crate::{bitcoin, monero};
+
+#[derive(Debug)]
+pub enum Message {
+    Message0(Message0),
+    Message1(Message1),
+    Message2(Message2),
+}
+
+#[derive(Debug)]
+pub struct Message0 {
+    pub(crate) A: bitcoin::PublicKey,
+    pub(crate) S_a_monero: monero::PublicKey,
+    pub(crate) S_a_bitcoin: bitcoin::PublicKey,
+    pub(crate) dleq_proof_s_a: cross_curve_dleq::Proof,
+    pub(crate) v_a: monero::PrivateViewKey,
+    pub(crate) redeem_address: bitcoin::Address,
+    pub(crate) punish_address: bitcoin::Address,
+}
+
+#[derive(Debug)]
+pub struct Message1 {
+    pub(crate) tx_cancel_sig: Signature,
+    pub(crate) tx_refund_encsig: EncryptedSignature,
+}
+
+#[derive(Debug)]
+pub struct Message2 {
+    pub(crate) tx_lock_proof: monero::TransferProof,
+}
+
+impl From<Message0> for Message {
+    fn from(m: Message0) -> Self {
+        Message::Message0(m)
+    }
+}
+
+impl TryFrom<Message> for Message0 {
+    type Error = UnexpectedMessage;
+
+    fn try_from(m: Message) -> Result<Self, Self::Error> {
+        match m {
+            Message::Message0(m) => Ok(m),
+            _ => Err(UnexpectedMessage {
+                expected_type: "Create0".to_string(),
```

Comment:
Stale string? Should probably say `Message0`. Could implement this as a macro and use `stringify!()` to get the name of the struct you are implementing it on.